src/modules/structure.js

```diff
@@ -0,0 +1,138 @@
+let tasks = [];
+const setLocalStorage = () => {
+  localStorage.setItem('task', JSON.stringify(tasks));
+};
+const getLocalStorage = () => (localStorage.getItem('task') ? JSON.parse(localStorage.getItem('task')) : []);
+
+tasks = getLocalStorage();
+const listContainer = document.querySelector('.all-tasks');
+
+const populateTasks = () => {
+  const listContainer = document.querySelector('.all-tasks');
+  listContainer.innerHTML = '';
+  const listElements = tasks.map((task) => {
+    const li = `<li class="list--item">
+    <div class="desc--container">
+        <input type="checkbox" name="" id="">
+        <div id=${task.index} class="parent--div">
+            <p class="desc">${task.desc}</p>
+            <form action="" id=${task.index} class="edit--form">
+                <input type="text" name="" class="edit--input" id="" >
+            </form>
+        </div>
+    </div>
+    <div class="icons--container">
+        <i class="fa-solid  fa-ellipsis-vertical" data-vertical=${task.index}></i>
+        <i class="fa-solid fa-trash-can" data-delete=${task.index} ></i>
+    </div>
+  </li>`;
+    return li;
+  }).join('');
+  listContainer.insertAdjacentHTML('beforeend', listElements);
+  const deleteIcons = listContainer.querySelectorAll('.fa-trash-can');
+  const verticalIcons = listContainer.querySelectorAll('.fa-ellipsis-vertical');
+  verticalIcons.forEach((icon) => {
+    icon.addEventListener('click', (e) => {
+      const { parentElement } = e.target.parentElement;
+      const editInput = parentElement.querySelector('.edit--input');
+      const descText = parentElement.querySelector('.desc');
+      parentElement.classList.add('about--to--edit');
+      editInput.value = descText.textContent;
+      editInput.focus();
+    });
+  });
+  deleteIcons.forEach((icon) => {
+    icon.addEventListener('click', (e) => {
+      const data = e.target.dataset.delete;
+      // eslint-disable-next-line no-use-before-define
```

Comment:
[ OPTIONAL ]
Great job so far. 🚀 There is one suggestion I would want to make here to improve your code. ♻️ 

- [ ] Kindly make sure you remove ` // eslint-disable-next-line no-use-before-define` because by enforcing declarations before usage, you are forced to consider the order of your code and ensure that necessary dependencies are declared and available when needed. 💯 

This promotes better design practices and can help catch potential issues earlier. ✅